### PR TITLE
assertion: add support for asserting on error `msg` and `status`

### PR DESCRIPTION
**Pull request by stephenmathieson**
see https://github.com/segmentio/integration-tester/pull/39#discussion_r107217979

@segmentio/pizza

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -567,7 +567,7 @@ Assertion.prototype.expects = function(){
 };
 
 /**
- * Assert that the integration errors with optional error `msg` or `status`.
+ * Assert that the integration errors with optional error `msg` and/or `status`.
  *
  * @param {String} [msg]
  * @param {Number} [status]
```
Comment (ndhoule) [inline lib/index.js]: I think these are backward now; in your tests below you have status then msg

File: test/assertion.js [hunk 2]
```diff
@@ -336,7 +336,7 @@ describe('Assertion', function(){
         .error('cannot POST /json/identify?baz=foo (400)', done);
     });
 
-    it('should throw if error messages dont match', function(done){
+    it('should throw if error messages does not match', function(done){
```
Comment (ndhoule) [inline test/assertion.js]: singular `messages` if you're actually concerned with "engrish" here. (there's probably a better commit message for this.)

Comment (stephenmathieson) [inline test/assertion.js]: I'll squash commits.. is this blocking? I'd like to move on now

Comment (ndhoule) [inline test/assertion.js]: You made the change :man_shrugging:

**Pull request merged**